### DISCUSSION TITLE
change_list template renders incorrectly when WSGI URL-path is not "/"

### DISCUSSION THREAD

**Discussion by ghost** [celery/django-celery-beat#470] [Q&A]
### Summary:

Suppose you have Django set up so the top level url is http://thehost/myapp/.  Then your URL root is "/myapp".  All the Django admin page templates then render correctly with links like,

`https://thehost/myapp/admin/auth/ `

But the links are wrong in pages that render with the django-celery-beat _change_list_ template (https://github.com/celery/django-celery-beat/blob/master/django_celery_beat/templates/admin/djcelery/change_list.html).

To be clear:  _All other Django admin pages render correctly_.  Only the django-celery-beat change list pages are broken.

The links on the celery beat "Change periodic task" page look like,

`https://thehost/admin/django_celery_beat/`

when the correct link should be,

`https://thehost/myapp/admin/django_celery_beat/
`

* Celery Version: 4.4.6
* Celery-Beat Version: 2.0.0

### Exact steps to reproduce the issue:
1. Configure the URL prefixing. In your apache WSGI configuration ensure that the URL-path contains more than "/".  Here I show it as "/myapp":

```
  WSGIScriptAlias /myapp /path/to/django/site/wsgi.py
```

2. Navigate to https://thehost/myapp/admin/django_celery_beat/periodictask/1/change/
3. Click on the _Periodic Tasks_ link at the top of the page (or just check the link with some html inspector).

**Comment by HIJADI** [comment]
May be not directly related to this issue, however, for fixing/changing the admin site "VIEW SITE" link for a Django app with non-root URL (eg /myapp) with apache and mod_wsgi:   override the admin site_url context, for example, see code below for overriding admin generate_site_url in apps.py:

```
from django.apps import AppConfig

class MyAppConfig(AppConfig):
    name = 'myApp'
    verbose_name = 'My App'

    def ready(self):
        from django.contrib import admin
        from django.contrib.admin import sites

        class MyAdminSite(admin.AdminSite):
            # Customize admin site
            site_header = 'My App Administration!'

            def each_context(self, request):
                context = super().each_context(request)
                context['site_url'] = self.generate_site_url(request)
                return context

            def generate_site_url(self, request):
                # eg fix site_url to / and not /myapp when through apache via request.META['SCRIPT_NAME']
                return '/'

        mysite = MyAdminSite()
        admin.site = mysite
        sites.site = mysite
```
